app.rb

```diff
@@ -0,0 +1,168 @@
+require_relative 'classes/book'
+require_relative 'classes/person'
+require_relative 'classes/teacher'
+require_relative 'classes/student'
+require_relative 'classes/rental'
+
+class App
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  def clear_screen
+    system('cls')
+    system('clear')
+  end
+
+  def create_teacher
+    puts '-creating a new teacher-'
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'specialization: '
+    spec = gets.chomp
+    @persons.push(Teacher.new(age, spec, name))
+    puts "Person created successfully\n"
+  end
+
+  def permission?
+    print "Has parents' permission? [Y/N]: "
+    option = gets.chomp.capitalize
+    case option
+    when 'N'
+      false
+    when 'Y'
+      true
+    else
+      permission?
+    end
+  end
+
+  def create_student
+    puts '-creating a new student-'
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+    permission = true
+    permission = permission? if age < 18
+    @persons.push(Student.new(age, name, parent_permission: permission))
+    puts "Person created successfully\n"
+  end
+
+  def create_person
+    clear_screen
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    option = gets.chomp.to_i
+    case option
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      create_person
+    end
+  end
```

Comment:
- [ ] @IsmailChangezi kindly check on your create person method, am getting the errors below when I try to create a student or a teacher, this made it impossible to test the create rental and display rental list.
![image](https://github.com/IsmailChangezi/oop-ruby/assets/61408860/35ab3a91-5990-480e-8bc0-093f51a17cef)
![image](https://github.com/IsmailChangezi/oop-ruby/assets/61408860/81d685c8-8465-43d7-a596-76942119b7a8)